### ISSUE TITLE
Configs: add `prefer-nullish-coalescing` to the `recommended` config group

### ISSUE THREAD

**Issue by Vinnl** [typescript-eslint/typescript-eslint#8144]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22preset+config+change%22+) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Description

`prefer-nullish-coalescing` is currently in the `stylistic` config group. However, it can catch actual bugs; for example, if providing a fallback value for a number for which 0 (or any falsy non-`null` non-`undefined` value) is a valid value:

```ts
// If the given age is `0`, this will be 18
const age = givenAge || 18;
```

In cases where a falsy value is not valid, either nullish coalescing will still work fine...

```ts
// If `filterText` is an empty string, it doesn't matter that `??` doesn't result in the fallback
const filter = filterText ?? "";
```

...or the falsy value can be matched explicitly:

```ts
const name = enteredName === "" ? "(Unknown)" : enteredName;
```

The latter case is somewhat opinionated, I suppose, so the `strict` sense might make sense instead. In any case, this isn't a stylistic rule, as far as I'm concerned.

### Impacted Configurations

The `stylistic` and either `recommended` or `strict` config groups.

**Comment by bradzacher** [comment]
The reason that it's stylistic is because the rule cannot delineate between intentional and unintentional cases - a non-trivial number of people use `||` intentionally because it coerces instead of strictly checks.

Whilst we do recommend the style - it's not something we can enforce via the recommended set. That set instead is specifically intended to be a lowest-common-denominator config which includes rules that we recommend AND that people won't turn off due to it being an opinionated style.

If you're looking for a rule that is there to catch and enforce against this kind of ambiguity and prevent such classes of bugs - [`strict-boolean-expressions`](https://typescript-eslint.io/rules/strict-boolean-expressions/) is a much better option.
